src/Client/WorkflowExecutionHistory.php

```diff
@@ -71,4 +71,22 @@ public function getIterator(): Traversable
     {
         return $this->getEvents();
     }
+
+    /**
+     * Stores workflow history to a file.
+     */
+    public function toFile(string $file)
+    {
+        // Check file
+        if (!\is_dir(\dirname($file))) {
+            throw new \RuntimeException(\sprintf('Directory "%s" does not exist.', \dirname($file)));
+        }
+
+        $events = \iterator_to_array($this->getEvents(), false);
+        $history = new History();
+        $history->setEvents($events);
+
+        $data = $history->serializeToJsonString();
+        \file_put_contents($file, $data);
+    }
```

Comment:
@Sushisource hi.
I'm trying to store Events History using protobuf to json serialization.

And I'm getting invalid JSON as result:

[history-files.zip](https://github.com/temporalio/sdk-php/files/12293461/history-files.zip)

All enums have different values:
![{9BE90F8F-C69D-4797-AA3E-B1C258BEB670}](https://github.com/temporalio/sdk-php/assets/4152481/a3b9decb-f285-4fab-9a11-5caaa0cb502a)

If we really need to have the ability to store history from PHP collection then I need to convert all these values in a right way.

Reply:
Well, at least properties named correctly.

Reply:
Yeah, so you'll need to use the Go SDK to deserialize such histories, because it uses a weird forked version of gogoproto that does this. It's quite obnoxious. I was thinking you were already going through Go for both the serialize and deserialize operations.

We could do something annoying like check the first event (always workflow execution started) and depending on how `eventType` looks, either deserialize via Go, or do it directly.

This would result in supporting downloaded histories from the Web UI too, which also get serialized the Go way, so that is probably desirable. Annoying, but should be fairly easy.

Reply:
We have already implemented the whole history download in both formats and replay on Golang's end. We have tested the ability to replay history from UI with the code that was exposed with these 3 methods.

Sounds like reading history on PHP is not a problem, we can probably also save it to protobuf file and replay, but I'm not sure why we need to convert it anyhow when all the de-serialization logic is on Golang end.

Reply:
We can also try to de-serialize whole history from pb to JSON using new RPC command instead of going entry by entry.